components/discord-window.tsx

```diff
@@ -0,0 +1,68 @@
+import Image from 'next/future/image'
+import Draggable from 'react-draggable'
+
+const DiscordWindow = ({
+  channel,
+  content,
+  pfpImageSrc,
+  username,
+  imageSrc
+}: {
+  channel: string
+  content: string
+  pfpImageSrc: string
+  username: string
+  imageSrc?: string
+}) => {
+  return (
+    <Draggable handle=".handle">
+      <div className="message border-2 border-black w-11/12 sm:w-96 shadow-email shadow-gray-900/30 h-fit z-0 relative odd:right-40 odd:sm:right-96 sm:odd:top-4 sm:even:bottom-8 even:left-40 even:sm:left-96 sm:first-of-type:right-80 sm:last-of-type:left-80">
+        <div className="flex flex-row bg-gray-discord-darker cursor-pointer text-gray-100">
+          <p className="px-2 border-r-2 border-b-2 border-black bg-red-700 hover:bg-red-600">
+            ∆
+          </p>
+          <div className="grow handle" />
+          <p className="handle">{channel}</p>
+          <div className="grow handle" />
+          <div />
+        </div>
+        <div className="border-black flex flex-col bg-gray-discord p-2 text-white text-sm">
+          <div className="flex flex-row gap-4">
+            <div>
+              <div className="w-10">
+                <Image
+                  src={pfpImageSrc}
+                  alt="Profile picture"
+                  width={0}
+                  height={0}
+                  sizes="100%"
+                  className="rounded-full w-auto h-auto"
+                />
+              </div>
+            </div>
+            <div className="flex flex-col">
+              <p className="text-gold-discord">{username}</p>
+              <p>{content}</p>
+              {imageSrc && (
+                <div className="w-1/2">
+                  <a href={imageSrc} target="_blank">
+                    <Image
+                      src={imageSrc}
+                      alt="Image"
+                      width={0}
+                      height={0}
+                      sizes="100%"
+                      className="rounded w-auto h-auto mt-1"
```

Comment:
I think you can probably replace width={0} and height = {0} with "layout='fill' objectFit='contain'" according to this post: https://stackoverflow.com/questions/65169431/how-to-set-the-next-image-component-to-100-height
(if that way doesn't work I think it's fine to keep it like this. tbh I don't even know if that is the right way lol)

Reply:
That appears to fill the entire div:

<img width="1122" alt="Screenshot 2022-12-06 at 11 14 53 PM" src="https://user-images.githubusercontent.com/14811170/206086821-6493cd61-0a0f-45aa-9064-3a0ece238fbf.png">

What I'm trying to do here is make it so that the width and height are dynamic and are sized by the div rather than by the image.

I'm probably missing something but idk it works :P